### PR TITLE
Use codecov-python

### PR DESCRIPTION
**Pull request by jayvdb**
Currently codecov-bash is used, which involves an uncached fetch
of a unversioned bash script.

Using codecov-python allows the tool to be cached, and greater
reliability due to versioned releases of the tool.

In addition, codecov-bash does not work on Windows, and our
developer base is better tuned to investigate any python
related problems.

Fixes https://github.com/coala/coala/issues/3718
Related to https://github.com/coala/coala-bears/issues/1415

### CODE REVIEW

**Comment by jayvdb** [comment]
This is moderately urgent as their has been a few failures due to fetching the bash script.
Most recently it hit me. https://travis-ci.org/coala/coala-bears/builds/236197559

File: .ci/appveyor.yml [hunk 1]
```diff
@@ -59,3 +59,12 @@ test_script:
...
   - "%CMD_IN_ENV% python setup.py install"
+
+on_success:
+  - codecov
+
+on_failure:
+  - codecov
+
+matrix:
+  fast_finish: true
```
Comment (underyx) [inline .ci/appveyor.yml]: Oh hey, what's this one? Shouldn't it be in a separate commit?

**Comment by yukiisbored** [comment]
ack https://github.com/coala/coala-bears/pull/1770/commits/0bfda9d0678d6e9fd1ea841042f383ed46a2e5ec

File: .ci/appveyor.yml (hunk 1 above)
Comment (jayvdb) [inline .ci/appveyor.yml]: It was also tacked onto a different commit at coala/coala@4f16f6f74

**Comment by jayvdb** [comment]
reack cfa562f87816bfeb170f7e1073643b49665465f3

**Comment by jayvdb** [comment]
@rultor merge
